<a href="https://colab.research.google.com/github/ardianita/data-science-2026/blob/main/Pertemuan_2_Ardianita_Fauziyah_NIM_250401020128.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
import numpy as np

# Muat Dataset Titanic
URL = "https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv"
df = pd.read_csv(URL)
df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


In [44]:
# Tipe Data Dataset Titanic
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


In [14]:
# Jumlah Kolom dan Baris Dataset Titanic
print(df.shape, df.columns.tolist())
print(f"Total Baris : {df.shape[0]}")
print(f"Total Kolom : {df.shape[1]}")
print(f"Kolom : {df.columns.tolist()}")

(891, 12) ['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked']
Total Baris : 891
Total Kolom : 12
Kolom : ['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked']


In [15]:
# Kolom yang memiliki nilai kosong
print(df.isnull().sum())

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64


Kolom yang memiliki nilai kosong : **Age**, **Cabin**, **Embarked**

In [40]:
# Filter wanita kelas 1
wk1 = df[(df['Sex'] == 'female') & (df['Pclass'] == 1)]
print(f"Total Wanita Kelas 1 Selamat : {wk1['Survived'].sum()}")
print(f"Presentase Wanita Kelas 1 Selamat : {(wk1['Survived'].mean() * 100).round(2)} %")

Total Wanita Kelas 1 Selamat : 91
Presentase Wanita Kelas 1 Selamat : 96.81 %


In [30]:
# Rata - rata usia wanita kelas 1
selamat = df[(df['Sex'] == 'female') & (df['Pclass']==1) & (df['Survived'] == 1)]
print(f"Rata - rata usia : {selamat['Age'].mean().round(1)} Tahun")

Rata - rata usia : 34.9 Tahun


In [39]:
# Kelas penumpang dengan tingkat keselamatan tertinggi
hasil = df.groupby('Pclass').agg(
    passenger=('PassengerId','count'),
    survived=('Survived','sum'),
    mean=('Survived','mean'),
    ).round(3)
hasil

,passenger,survived,mean
Pclass,,,
1,216,136,0.630
2,184,87,0.473
3,491,119,0.242


Kelas dengan tingkat keselamatan tertinggi adalah **Kelas 1** dengan total penumpang yang selamat sebanyak **136 orang** atau setara dengan **63%** dari total penumpang.

In [43]:
# Rata - rata tingkat keselamatan per kelas
persentase = df.groupby('Pclass')['Survived'].mean()
persentase.round(3)

,Survived
Pclass,
1,0.630
2,0.473
3,0.242
